wikimedia/WikiContrib
# Count Contributions to Wikimedia Accounts On Github

Prior to implementing this feature, we weren't accounting for user contributions to Wikimedia accounts on github. 

Things implemented:
 * This feature makes use of the GitHub commits search rest API endpoint to fetch contributions in form of commits to the master branch. 
 * Because the commits search endpoint is still in preview, I also implemented a fallback that fetches contributions as PRs instead once "GITHUB_FALLBACK_TO_PR" variable in settings is set to true, this is to account for the unlikely occasion that GitHub makes a breaking change to the commits search rest API endpoint.
 * Because of the nature of commits (commits are always greater than PRs), and the fact that Github search rest API endpoints have strict rate limits and other restrictions, this feature can only fetch one user's contributions per minute.  we had to implement a feature that ensures that a user is alerted once they hit the rate limit and advised on when next to try again.


**To Test**
You need to obtain your Github access token and add it to the .env file. Follow [this link](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) on how to do that. After you obtain the access token, look at the .env.example file to see how it should look inside the .env file
